letter-sender/src/main/java/com/christmas/letter/sender/service/SnsService.java

```diff
@@ -0,0 +1,53 @@
+package com.christmas.letter.sender.service;
+
+import com.christmas.letter.sender.exceptions.InvalidDataException;
+import com.christmas.letter.sender.model.Letter;
+
+import java.util.regex.Pattern;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.PublishRequest;
+
+@Service
+@RequiredArgsConstructor
+public class SnsService {
```

Comment:
The fact that this service uses SNS is an implementation detail. Let's find a more suitable name for it, describing what it does, not how it does it.

Reply:
changed to LetterService